### DISCUSSION TITLE
Alter column to nullable

### DISCUSSION THREAD

**Discussion by bernardoveras** [simolus3/drift#1969] [Q&A] [answered]
I want to turn an already existing column into NULLABLE, how do I add this change to the migration?

**Comment by simolus3** [comment, marked as answer]
This is quite simple to do with drift's builtin support for [complex table migrations](https://drift.simonbinder.eu/docs/advanced-features/migrations/#changing-column-constraints). Essentially, you

1. Change your table definition to make the column non-nullable.
2. Re-run the build runner and increase your `schemaVersion`.
3. Write a migration that calls `await m.alterTable(TableMigration(yourModifiedTable));`. This will re-create the table without loosing data. In your case, you don't need a more complex migration since all the existing data is compatible with the new schema (non-nullable rows can of course be stored in a nullable column).

**Reply by bernardoveras** [reply]
Thanks bro!
